client/src/components/Panels/utilities.js

```diff
@@ -98,18 +98,56 @@ export function searchToolsByKeys(tools, keys, query) {
                 actualValue = tool[key] ? tool[key].toLowerCase() : "";
             }
             const queryLowerCase = query.trim().toLowerCase();
+            // do we care for exact matches && is it an exact match ?
+            const order = keys.exact && actualValue === queryLowerCase ? keys.exact : keys[key];
             if (actualValue.match(queryLowerCase)) {
-                // do we care for exact matches && is it an exact match ?
-                const order = keys.exact && actualValue === queryLowerCase ? keys.exact : keys[key];
                 returnedTools.push({ id: tool.id, order });
                 break;
             }
+            else if (key !== "combined" && queryLowerCase.length >= 5 && dLDistance(queryLowerCase, actualValue)) {
+                returnedTools.push({ id: tool.id, order });
+            }
         }
     }
     // sorting results by indexed order of keys
     return orderBy(returnedTools, ["order"], ["desc"]).map((tool) => tool.id);
 }
 
+export function dLDistance(query, value) {
```

Comment:
Instead of reimpmementing this here, I'd probably suggest consolidation with https://github.com/galaxyproject/galaxy/blob/dev/client/src/utils/levenshtein.js

Reply:
I'll look in to updating the levensnshtein.js function and will possibly add the new Damerau–Levenshtein distance function if it isn't there. Also if someone has the time I changed the functionality of the search again to better deal with substring partial matches. Ie) if someone misspells "filtar" filter will still show up and all results that also contain the substring filter.